client/src/main/java/io/pravega/client/stream/impl/ControllerResolverFactory.java

```diff
@@ -121,10 +143,10 @@ public String getServiceAuthority() {
         @Override
         @Synchronized
         public void start(Listener listener) {
```

Comment:
This is not a general purpose class. This is an implementation of an interface which has to be registered with gRPC, which internally manages the lifecycle of this class. Not sure why we need to make this a guava service.